amespahdbpythonsuite/transitions.py

```diff
@@ -350,24 +668,54 @@ def cascade(self, e: float, **keywords) -> None:
                 global intensities
                 intensities = np.array([d["intensity"] for d in self.data[uid]])
 
-                Tmax = optimize.brentq(self.attainedtemperature, 2.73, 5000.0)
+                if keywords.get('approximate'):
+                    totalcrossection = np.sum(intensities)
 
+                Tmax = optimize.brentq(func1, 2.73, 5000.0)
+                self.model["temperatures"][uid] = Tmax
                 print("MAXIMUM ATTAINED TEMPERATURE     : %f Kelvin" % Tmax)
 
-                self.model["temperatures"].append({"uid": uid, "temperature": Tmax})
-
-                for d in self.data[uid]:
-                    if d["intensity"] > 0:
-                        global frequency
-                        frequency = d["frequency"]
-                        d["intensity"] *= (
-                            d["frequency"] ** 3
-                            * integrate.quad(self.featurestrength, 2.73, Tmax)[0]
-                        )
+                if (keywords.get('star') or keywords.get('isrf')) and keywords.get(
+                    'convolved'
+                ):
+                    for d in self.data[uid]:
+                        if d["intensity"] > 0:
+                            frequency = d["frequency"]
+                            d["intensity"] *= (
+                                d["frequency"] ** 3
+                                * integrate.quad(func3, 2.5e3, 1.1e5)[0]
+                            )
+                    intensities = np.array([x / Nphot for x in intensities])
+
+                else:
+                    for d in self.data[uid]:
+                        if d["intensity"] > 0:
+                            frequency = d["frequency"]
+                            d["intensity"] *= (
+                                d["frequency"] ** 3
+                                * integrate.quad(func2, 2.73, Tmax)[0]
+                            )
 
                 i += 1
 
-            print(57 * "=")
+            if not keywords.get('approximate'):
+                intensities = np.array(
+                    [x * y**3 for x, y in zip(intensities, frequencies)]
+                )
+            else:
+                intensities = np.array(
+                    [
+                        x * 2.48534271218563e-23 * nc * y**3 / totalcrossection
+                        for x, y in zip(intensities, frequencies)
+                    ]
+                )
+
+            print(
+                "ENERGY CONSERVATION IN SPECTRUM "
+                f" :{np.sum(intensities) / self.model['energy'][uid]['e']}"
+            )
+
+        print(57 * "=")
 
         elapsed = timedelta(seconds=(time.perf_counter() - tstart))
         print(f"Elapsed time: {elapsed}\n")
```

Comment:
Use `message` here. Of course, will need to have a better progress monitor at some point.